File: library-domain/src/main/java/com/dodecaedro/library/domain/repository/FineRepository.java
package com.dodecaedro.library.domain.repository;

import com.dodecaedro.library.domain.model.Borrow;
import com.dodecaedro.library.domain.model.Fine;
import com.dodecaedro.library.domain.model.User;

import java.time.ZonedDateTime;
import java.util.List;
import java.util.Optional;

public interface FineRepository {
	Optional<Fine> findTopByUserOrderByFineEndDateDesc(User user);

	Fine save(Fine fine);

	List<Fine> findActiveFinesInDate(User user, ZonedDateTime time);

	Optional<Fine> findByBorrow(Borrow borrow);
}

File: library-domain/src/main/kotlin/com/dodecaedro/library/domain/repository/FineRepository.kt
package com.dodecaedro.library.domain.repository

import com.dodecaedro.library.domain.model.Borrow
import com.dodecaedro.library.domain.model.Fine
import com.dodecaedro.library.domain.model.User
import java.time.ZonedDateTime
import java.util.*

interface FineRepository {
	fun findTopByUserOrderByFineEndDateDesc(user: User): Fine?

	fun save(fine: Fine): Fine

	fun findActiveFinesInDate(user: User, time: ZonedDateTime): List<Fine>

	fun findByBorrow(borrow: Borrow): Fine?
}
